quadriga/src/main/java/edu/asu/diging/quadriga/web/EditCollectionController.java

```diff
@@ -0,0 +1,101 @@
+package edu.asu.diging.quadriga.web;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.quadriga.core.model.Collection;
+import edu.asu.diging.quadriga.core.service.CollectionManager;
+import edu.asu.diging.quadriga.web.forms.CollectionForm;
+
+/**
+ * This controller enables users to edit their collections
+ * 
+ * @author poojakulkarni
+ *
+ */
+@Controller
+public class EditCollectionController {
+
+    @Autowired
+    private CollectionManager collectionManager;
+
+    /**
+     * Request handler for getting the "Edit collections" view
+     * 
+     * @param id    of the collection to be edited
+     * @param model to set collectionForm attribute to pre-fill current name and
+     *              description
+     * @return the editCollection view
+     */
+    @RequestMapping(value = "/auth/collections/edit", method = RequestMethod.GET)
+    public String get(@RequestParam(value = "id", required = true) String id, Model model) {
+        Optional<Collection> collection = Optional.ofNullable(collectionManager.findCollection(id));
+
+        if (collection.isPresent()) {
+            Collection currentCollection = collection.get();
+            CollectionForm collectionForm = new CollectionForm();
+            collectionForm.setId(id);
+            collectionForm.setName(currentCollection.getName());
+            collectionForm.setDescription(currentCollection.getDescription());
+            model.addAttribute("collectionForm", collectionForm);
+
+            return "auth/editCollection";
+        }
+
+        return "redirect:/auth/collections";
```

Comment:
there should be a 404 not found page shown to the user if the collection doesn't exist (not the tomcat one but a custom quadriga 404 one).

Reply:
Should I create one or does one exist already?

Reply:
You have to check. There probably is not yet one.